pkg/controller/registry/reconciler/grpc.go

```diff
@@ -196,16 +198,25 @@ func (c *GrpcRegistryReconciler) currentPodsWithCorrectImageAndSpec(source grpcC
 	found := []*corev1.Pod{}
 	newPod := source.Pod(saName)
 	for _, p := range pods {
-		if p.Spec.Containers[0].Image == source.Spec.Image && podHashMatch(p, newPod) {
+		if correctImages(source, p) && podHashMatch(p, newPod) {
```

Comment:
I know you're just updating based on what's already here, but I do wonder if `podHashMatch` alone would be sufficient here. Wouldn't that cover the image references in the pod spec?

Reply:
I don't have enough knowledge of the system to understand why that was not the case before, so I did not want to change it. Perhaps a follow-up? Knowing this code-base it will break some obscure edge cases and be hard to track down :)